app/src/main/java/com/strv/movies/extension/AssistedViewModel.kt

```diff
@@ -0,0 +1,20 @@
+package com.strv.movies.extension
+
+import androidx.activity.ComponentActivity
+import androidx.activity.viewModels
+import androidx.lifecycle.AbstractSavedStateViewModelFactory
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+
+@Suppress("UNCHECKED_CAST")
+inline fun <reified T : ViewModel> ComponentActivity.assistedViewModel(
+    crossinline viewModelProducer: (SavedStateHandle) -> T
+) = viewModels<T> {
+    object : AbstractSavedStateViewModelFactory(this, intent.extras) {
+        override fun <T : ViewModel> create(
+            key: String,
+            modelClass: Class<T>,
+            handle: SavedStateHandle
+        ) = viewModelProducer(handle) as T
+    }
+}
```

Comment:
This seems decently hardcore, maybe you can just give this piece of code to your students.